keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,835 @@
+// on its attributes. Besides the name, exactly one field must be set.
+// +k8s:deepcopy-gen=true
+type Device struct {
+	// Name is unique identifier among all devices managed by
+	// the driver in the pool. It must be a DNS label.
+	//
+	// +required
+	Name string `json:"name"`
+	...
+
+	// ExtendedResourceName is the extended resource name
+	// the device is advertised as. It must be a DNS label.
+	// It overrides the ExtendedResourceName at ResourceSlice if both are
+	// present.
+	//
+	// +optional
+	ExtendedResourceName string
+}
+```
+
+The devices can be advertised with an extended resource name. The extended
+resource name can be specified on each individual device. Different
+devices can be advertised as different extended resource name, or not
+advertised as extended resource at all.
+
+Alternatively, the extended resource name can be specified at the
+`ResourceSlice` level, then all the devices in the resource slice are
+advertised as the given extended resource name. If a device has a different
+extended resource name than that given in the `ResoureSlice`, the device's
+extended resource name is used for that device.
+
+### Resource Claim API
+There is no API change on `ResourceClaim`, i.e. no new API type. However, a special
+resource claim object is created to keep track of device allocations for dyanmic
+extended resource. The special resource claim object has following properties:
+
+  * It is namespace scoped, like other resource claim objects.
+  * It is owned by a pod, like other resource claim objects.
+  * It has null `spec`.
+  * Its `status.allocation.devices` and `status.allocation.reservedFor` are
+    used.
+  * It has annotation `resource.kubernetes.io/extended-resource-name:`, and it
+    does not have annotation `resource.kubernetes.io/pod-claim-name:`
+
+```yaml
+metadata:
+  annotations:
+    resource.kubernetes.io/extended-resource-name: foo.domain/bar
+```
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic extended resource plugin during
+    pre-bind phase. The in-memory one in the assumed cache is created earlier
+    during reserve phase.
+  * It is *deleted* together with the owning pod's deletion.
+  * It is *read* by scheduler dyanmic resource plugin for the devices allocated,
+  so that the scheduler remove considerations for allocation of these devices for
+  other DRA resource claim requests in 'dynamic resource plugin'.
+  * It is *read* by the kubelet DRA device driver to prepare the devices listed
+    therein when preparing to run the pod.
+
+### Pod API
+There is no API change on `Pod`. Pod's status.resourceClaimStatuses tracks the
```

Comment:
There is: relaxing validation of `name` is an API change. `name: foo.domain/bar` should be invalid at the moment (not a DNS label, not the name of an entry in pod.spec.resourceClaims).

Reply:
as discussed in another comment, we probably don't want to relax the validation of name, to use it for extended resource name. it's cleaner to have a separate pod.status.extendedResourceClaimStatus.